tree/ntuple/v7/inc/ROOT/RNTupleProcessor.hxx

```diff
@@ -264,6 +298,65 @@ private:
    RNTupleChainProcessor(const std::vector<RNTupleOpenSpec> &ntuples, std::unique_ptr<RNTupleModel> model = nullptr);
 };
 
+// clang-format off
+/**
+\class ROOT::Experimental::RNTupleJoinProcessor
+\ingroup NTuple
+\brief Processor specializiation for horizontally concatenated RNTuples (joins).
+*/
+// clang-format on
+class RNTupleJoinProcessor : public RNTupleProcessor {
+   friend class RNTupleProcessor;
+
+private:
+   std::unique_ptr<RNTupleModel> fJoinModel;
+   /// Mapping of the auxiliary RNTuple name to its page source
+   std::unordered_map<std::string, std::unique_ptr<Internal::RPageSource>> fAuxiliaryPageSources;
```

Comment:
Just for my own understanding, this data member implies that the horizontal concatenation can happen at the single-file-level, i.e. the right parts of the join cannot be vertically concatenated (at least for now), is this correct?

Reply:
For now, yes. I am currently (in parallel to this PR) working on the composition of the different concatenations, but I want to take this step by step to try and keep the PRs and code changes somewhat manageable.

Reply:
Very good thanks!